bin/run_docker.py

```diff
@@ -267,8 +268,55 @@ def mount_volumes() -> dict:
     return volumes
 
 
+def monitor_container(
+    container: docker.models.containers.Container,
+    timeout: Union[int, float],
+    poll_interval: Union[int, float],
+    elapsed_time: int = 0,
+):
+    """
+    Recursively monitor the status of a Docker container until it finishes,
+    or the timeout is reached, at which point the container is killed.
+
+    Arguments:
+        container: The Docker container subject to monitor.
+        timeout: Maximum duration to monitor the container (in minutes) before forced shutdown.
+        poll_interval: Time to wait between status checks (in minutes).
+        elapsed_time: The elapsed time already spent monitoring (in minutes).
+    """
+
+    # Check if the elapsed time has reached or exceeded the timeout
+    if elapsed_time >= timeout:
+        print("Timeout reached. Stopping container.")
+        container.kill()
```

Comment:
Stop then kill.  Look at the existing challengeWorkflowtemplates code here for an example